src/compas_timber/connections/l_halflap.py

```diff
@@ -62,43 +110,100 @@ def add_extensions(self):
             If the extension could not be calculated.
 
         """
-        assert self.main_beam and self.cross_beam
+        assert self.beam_a and self.beam_b
+        start_a, start_b = None, None
         try:
-            main_cutting_frame = self.get_main_cutting_frame()
-            cross_cutting_frame = self.get_cross_cutting_frame()
+            start_a, end_a = self.beam_a.extension_to_plane(self.cutting_plane_a.to_plane())
+            start_b, end_b = self.beam_b.extension_to_plane(self.cutting_plane_b.to_plane())
+        except AttributeError as ae:
+            # I want here just the plane that caused the error
+            geometries = [self.cutting_plane_b] if start_a is not None else [self.cutting_plane_a]
```

Comment:
normally I don't care much for performance, but it's worth paying attention when some properties are called multiple times where the value shouldn't have changed and the calculation can be rather signification (like in this case the `self.cutting_plane_*`)